src/mozillavpn.cpp

```diff
@@ -662,36 +681,19 @@ void MozillaVPN::removeDeviceFromPublicKey(const QString& publicKey) {
 
   const Device* device =
       m_private->m_deviceModel.deviceFromPublicKey(publicKey);
-  if (device) {
-    // Let's inform the UI about what is going to happen.
-    emit deviceRemoving(publicKey);
-    TaskScheduler::scheduleTask(new TaskRemoveDevice(publicKey));
-
-    if (m_state != StateDeviceLimit) {
-      // To have a faster UI, we inform the device-model that this public key
-      // is going to be removed.
-      m_private->m_deviceModel.startDeviceRemovalFromPublicKey(publicKey);
-    }
-  }
-
-  if (m_state != StateDeviceLimit) {
+  if (!device) {
     return;
   }
 
-  // Let's recover from the device-limit mode.
-  Q_ASSERT(!m_private->m_deviceModel.hasCurrentDevice(keys()));
-
-  // Here we add the current device.
-  addCurrentDeviceAndRefreshData();
-
-  // Finally we are able to activate the client.
-  TaskScheduler::scheduleTask(new TaskFunction([this]() {
-    if (m_state != StateDeviceLimit) {
-      return;
-    }
+  // Let's inform the UI about what is going to happen.
+  emit deviceRemoving(publicKey);
+  TaskScheduler::scheduleTask(new TaskRemoveDevice(publicKey));
 
-    maybeStateMain();
-  }));
+  if (m_state != StateDeviceLimit) {
+    // To have a faster UI, we inform the device-model that this public key
+    // is going to be removed.
```

Comment:
At which point is the UI informed about the device removal, here, on line 688 or both? And if both, why? This is confusing.